importers/network/network_importer_test.go

```diff
@@ -0,0 +1,118 @@
+package networkimporter
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/algorand/indexer/importers"
+	"github.com/algorand/indexer/plugins"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	logger       *logrus.Logger
+	ctx          context.Context
+	cancel       context.CancelFunc
+	testImporter importers.Importer
+)
+
+func init() {
+	logger = logrus.New()
+	logger.SetOutput(os.Stdout)
+	logger.SetLevel(logrus.InfoLevel)
+	ctx, cancel = context.WithCancel(context.Background())
+}
+
+func TestImporterorterMetadata(t *testing.T) {
+	testImporter = New()
+	metadata := testImporter.Metadata()
+	assert.Equal(t, metadata.Type(), networkImporterMetadata.Type())
+	assert.Equal(t, metadata.Name(), networkImporterMetadata.Name())
+	assert.Equal(t, metadata.Description(), networkImporterMetadata.Description())
+	assert.Equal(t, metadata.Deprecated(), networkImporterMetadata.Deprecated())
+}
+
+func TestCloseSuccess(t *testing.T) {
+	testImporter = New()
+	_, err := testImporter.Init(ctx, plugins.PluginConfig(
+		"genesis-path: /Users/ganesh/go-algorand/installer/genesis/mainnet/genesis.json"+
+			"\nconfig-path: /Users/ganesh/go-algorand/installer/"), logger)
```

Comment:
This might be a little more readable if it created the ImporterConfig object and serialized it to a string.

Also, I wonder how we could do this without depending on a genesis.json file.